controllers/nodehealthcheck_controller.go

```diff
@@ -398,11 +405,19 @@ func (r *NodeHealthCheckReconciler) isHealthy(conditionTests []remediationv1alph
 		if !exists {
 			continue
 		}
-		if n.Status == c.Status && currentTime().After(n.LastTransitionTime.Add(c.Duration.Duration)) {
-			return false
+		if n.Status == c.Status {
+			now := currentTime()
+			if now.After(n.LastTransitionTime.Add(c.Duration.Duration)) {
+				// unhealthy condition duration expired, node is unhealthy
+				return false, nil
+			} else {
+				// unhealthy condition duration not expired yet, node is healthy. Requeue when duration expires
+				expiresAfter := n.LastTransitionTime.Add(c.Duration.Duration).Sub(now) + 1*time.Second
```

Comment:
Why we add one second? Rule of thumb for a minimum (and reasonable) time quantity?

Reply:
good question , not sure if it's really needed, just want to ensure that we requeue when the duration expired for sure